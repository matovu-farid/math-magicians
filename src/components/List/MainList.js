import { Component } from 'react';
import propTypes from 'prop-types';
import BottomRow from './Rows/bottomrow';
import TopSymbolRow from './Rows/toprow';
import NumberList from './NumberList';

class MainList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { handleButtonClick } = this.props;
    return (
      <ul className="main-list">
        <TopSymbolRow handleButtonClick={handleButtonClick} />
        <NumberList handleButtonClick={handleButtonClick} />
        <BottomRow handleButtonClick={handleButtonClick} />
      </ul>
    );
  }
}

MainList.propTypes = {
  handleButtonClick: propTypes.func.isRequired,
};

export default MainList;
