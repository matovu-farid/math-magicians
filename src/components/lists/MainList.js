import { Component } from 'react';
import BottomRow from '../rows/bottomrow';
import TopSymbolRow from '../rows/toprow';
import NumberList from './NumberList';

class MainList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <ul className="main-list">
        <TopSymbolRow />
        <NumberList />
        <BottomRow />
      </ul>
    );
  }
}

export default MainList;
