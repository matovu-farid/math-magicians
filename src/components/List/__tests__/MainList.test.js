import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import MainList from '../MainList';

it('should have 14 buttons in total', () => {
  render(<MainList />);
  const lists = screen.queryAllByRole('listitem');

  expect(lists).toHaveLength(14);
});
