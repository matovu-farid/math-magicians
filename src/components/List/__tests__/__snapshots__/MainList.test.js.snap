// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="main-list"
>
  <ul
    className="flex"
  >
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </li>
  </ul>
  <ul
    className="num-list"
  >
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </li>
  </ul>
  <ul
    className="top-list"
  >
    <li>
      <button
        className="tile  wide"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </li>
    <li>
      <button
        className="tile"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </li>
  </ul>
</ul>
`;
